std-bits/table/src/main/java/org/enso/table/data/column/operation/map/numeric/DoubleLongBooleanOpWithSpecialNumericHandling.java

```diff
@@ -0,0 +1,63 @@
+package org.enso.table.data.column.operation.map.numeric;
+
+import org.enso.table.data.column.operation.map.TernaryMapOperation;
+import org.enso.table.data.column.operation.map.MapOperationProblemBuilder;
+import org.enso.table.data.column.storage.numeric.DoubleStorage;
+import org.enso.table.data.column.storage.Storage;
+import org.enso.table.error.UnexpectedTypeException;
+import org.graalvm.polyglot.Context;
+
+import java.util.BitSet;
+
+/** An operation expecting a numeric argument and returning a number. */
+public abstract class DoubleLongBooleanOpWithSpecialNumericHandling extends TernaryMapOperation<Double, DoubleStorage> {
+
+    public DoubleLongBooleanOpWithSpecialNumericHandling(String name) {
+        super(name);
+    }
+
+    protected abstract double doLongBoolean(double a, long b, boolean c, int ix, MapOperationProblemBuilder problemBuilder);
+
+    @Override
+    public Storage<Double> runTernaryMap(DoubleStorage storage, Object arg0, Object arg1, MapOperationProblemBuilder problemBuilder) {
+        if (arg0 == null || arg1 == null) {
+            return DoubleStorage.makeEmpty(storage.size());
+        }
+
+        long longArg;
+        if (arg0 instanceof Long) {
+            longArg = (Long) arg0;
```

Comment:
We can use the new syntax that does instanceof+cast in one operation, I think it is preferred.
```suggestion
        if (arg0 instanceof Long arg0AsLong) {
            longArg = arg0AsLong;
```

Reply:
Done.